src/auth/googleauth.ts

```diff
@@ -634,4 +643,63 @@ export class GoogleAuth {
           if (callback) callback(null, body);
         });
   }
+
+
+  /**
+   * Returns the client_email and private_key (if exists).
+   * It checks for these values from the user JSON at first.
+   * If it doesn't exist, and the environment is on GCE, it gets the
+   * client_email from the cloud metadata server.
+   * @param {function} callback Callback.
+   * @return object representation of the service account JSON
```

Comment:
* The return annotation isn't correct. The value is returned via the callback. Remove the return annotation and expand the callback instead.
* With that in mind, why is the callback optional? Make it required and remove all if `if (callback)` checks below.  I know the rest of this file accepts optional callbacks, but we can't fix those without a semver major change. This is a new API, so let's get this right the first time around. /cc @JustinBeckwith 
* Don't provide types in jsdoc when working with TypeScript. The types are already available in the signature.  Remove `{function} from the jsdoc for `callback`.

Reply:
Changed.